django/contrib/gis/db/backends/postgis/pgraster.py

```diff
@@ -129,7 +126,7 @@ def to_pgraster(rast):
 
         # Set the nodata flag
         if band.nodata_value is not None:
-            pixeltype += 64
+            pixeltype |= BANDTYPE_FLAG_HASNODATA
```

Comment:
And accordingly.
```python
# Encode pixeltype into 4 bits with leading zeros.
pixeltype = format(pixeltype, '04b')
# Set flags, is_offdb, is_nodata, and reserved flags always set to zero.
if band.nodata_value is not None:
    # Turn has_nodata flag on.
    flags = '0100'
else:
    flags = '0000'
# Convert to final integer number.
pixeltype = int(pixeltype + flags, 2)

Reply:
I'm not sure this is more readable than `pixeltype |= BANDTYPE_FLAG_HASNODATA` or `pixeltype += 64`.

Reply:
You are right here as well. Your version is clearer for people that understand bitwise operators the flags structure, plus its shorter. For me he more "explicit" version is easier to follow, and probably true too for other people that do not have experience with the postgis bites structure or bitwise operators. But the conciseness of your version is convincing so lets use it.